lib/parser/classes/NavigationEndpoint.ts

```diff
@@ -2,16 +2,41 @@ import Parser, { ParsedResponse } from "../index.js";
 
 // TODO: refactor this
 import { YTNode } from "..";
-import Actions from "../../core/Actions.js";
+import Actions, { ActionsResponse } from "../../core/Actions.js";
+import { ParsingError } from "../../utils/Utils.js";
 
 class NavigationEndpoint extends YTNode {
+    payload;
+    dialog;
+    metadata: {
+        url?: string;
+        api_url?: string;
+        page_type?: string;
+        send_post?: boolean; // TODO: is this boolean?
+    };
+    // TODO: these should be given proper types, currently infered
```

Comment:
Pretty sure we will be getting rid of those soon. The “`callTest()`” experiment is much more flexible than implementing every single endpoint manually (which would eventually end up creating one big chungus class). So no need to worry about giving them proper types IMHO.